src/util/queryNodeMatchers.ts

```diff
@@ -0,0 +1,147 @@
+import { Position, Range } from "vscode";
+import { SyntaxNode, Query, Language, QueryCapture } from "web-tree-sitter";
+import {
+  NodeMatcher,
+  NodeMatcherValue,
+  SelectionExtractor,
+  SelectionWithEditor,
+} from "../typings/Types";
+import { simpleSelectionExtractor } from "./nodeSelectors";
+
+export function defaultMatcher(
+  scopeType: string,
+  searchScopePresent: boolean,
+  scopeQuery: string,
+  selector: SelectionExtractor = simpleSelectionExtractor
+): NodeMatcher {
+  let query: Query;
+
+  return (
+    selection: SelectionWithEditor,
+    node: SyntaxNode,
+    siblings: boolean | undefined
+  ): NodeMatcherValue[] | null => {
+    if (!query) {
+      query = getQuery(node, scopeQuery);
+    }
+
+    const captures = extractCaptures(
+      selection,
+      node.tree.rootNode,
+      query,
+      scopeType
+    );
+
+    if (captures.length === 0) {
+      return null;
+    }
+    let capture: QueryCapture;
+    capture = selectCaptureByRange(captures, selection);
+    if (siblings) {
+      let captures: QueryCapture[];
+      if (searchScopePresent) {
```

Comment:
I think you mean 
```suggestion
      if (iterationScopePresent) {
```

The `searchScope` has nothing to do with iteration. It's used for thinks like the key key-value pairs: the pair is the search scope so that we can select the key from the value